synapse/models/person.py

```diff
@@ -1,41 +1,238 @@
-import synapse.lib.tufo as s_tufo
+import synapse.lib.types as s_types
 import synapse.lib.module as s_module
-from synapse.lib.types import DataType
 
-# FIXME identify/handle possibly as seeds
-# tony stark
-# tony logan stark
-# stark,tony logan
-# stark,tony l.
-
-class Name(DataType):
-    subprops = (
-        ('sur', {'ptype': 'ps:tokn'}),
-        ('given', {'ptype': 'ps:tokn'}),
-        ('middle', {'ptype': 'ps:tokn'}),
-        ('parts', {'ptype': 'int', 'doc': 'Number of ps:tokn elements in ps:name'}),
-    )
-
-    def norm(self, valu, oldval=None):
-        subs = {}
-
-        valu = valu.lower().strip()
-        if not valu:
-            self._raiseBadValu(valu)
-
-        parts = [v.strip().strip('.').lower() for v in valu.split(',')]
-        if len(parts) >= 2:
-            subs['sur'] = parts[0]
-            subs['given'] = parts[1]
-            if len(parts) >= 3:
-                subs['middle'] = parts[2]
-
-        subs['parts'] = len(parts)
-        valu = ','.join(parts)
-        return valu, subs
+class PsModule(s_module.CoreModule):
+    def getModelDefs(self):
+        modl = {
+            'types': (
+                ('ps:tokn', ('str', {'lower': True, 'strip': True}), {
+                    'doc': 'A single name element (potentially given or sur).',
+                    'ex': 'robert'
+                }),
+                ('ps:name', ('str', {'lower': True, 'onespace': True}), {
+                    'doc': 'An arbitrary, lower spaced string with normalized whitespace.',
+                    'ex': 'robert grey'
+                }),
+                # FIXME - 00x ps model supported guid alias 'ps:person:guidname'
+                ('ps:person', ('guid', {}), {
+                    'doc': 'A GUID for a person.',
+                }),
+                # FIXME - 00x ps model supported guid alias 'ps:persona:guidname'
+                ('ps:persona', ('guid', {}), {
+                    'doc': 'A GUID for a suspected person.',
+                }),
+                ('ps:person:has', ('comp', {'fields': (('person', 'ps:person'), ('node', 'ndef'))}), {
+                    'doc': 'A person owns, controls, or has exclusive use of an object or'
+                           ' resource, potentially during a specific period of time.'
+                }),
+                ('ps:persona:has', ('comp', {'fields': (('persona', 'ps:persona'), ('node', 'ndef'))}), {
+                    'doc': 'A persona owns, controls, or has exclusive use of an object or'
+                           ' resource, potentially during a specific period of time.'
+                }),
+                # FIXME requires file:bytes
```

Comment:
file:bytes is in 010 now

Reply:
i want to add file:bytes when we've stablized on what file:bytes is going to be